exist-core/src/main/java/org/exist/xquery/functions/validation/Jaxp.java

```diff
@@ -345,14 +378,47 @@ private void setXmlReaderFeature(XMLReader xmlReader, String featureName, boolea
         }
     }
 
-    private void setXmlReaderEnitityResolver(XMLReader xmlReader, XMLEntityResolver entityResolver ){
-
+    // TODO(AR) remove this when PR https://github.com/xmlresolver/xmlresolver/pull/98 is merged
+    private String serializeDocument(final XmldbURI documentUri) throws SAXException, IOException {
+        DocumentImpl doc = null;
         try {
-            xmlReader.setProperty(XMLReaderObjectFactory.APACHE_PROPERTIES_ENTITYRESOLVER, entityResolver);
+            doc = context.getBroker().getXMLResource(documentUri, Lock.LockMode.READ_LOCK);
+            if (doc == null) {
+                throw new IOException("No such document: " + documentUri);
+            }
 
-        } catch (final SAXNotRecognizedException | SAXNotSupportedException ex) {
-            LOG.error(ex.getMessage());
+            try (final StringWriter stringWriter = new StringWriter()) {
+                final Properties outputProperties = new Properties();
+                outputProperties.setProperty(OutputKeys.METHOD, "XML");
+                outputProperties.setProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+                outputProperties.setProperty(OutputKeys.INDENT, "no");
+                outputProperties.setProperty(OutputKeys.ENCODING, "UTF-8");
```

Comment:
ehm... didn't we agree on using `UTF_8.name()`

Reply:
Yes we did, I missed that one. I have now corrected it :-)

Reply:
Yeah, on the other hand I totally missed that this targets 4.x.x where the code still has the literal all over the place.